### PR TITLE
Refactor to use generic pub/sub from rclcpp

### LINKED ISSUE #28: Use generic pub/sub from rclcpp
Now that implementation for generic publishers and subscriptions exists in rclcpp, we can remove the duplicate code from domain_bridge.

### PR DESCRIPTION
**Pull request by rebecca-butler**
Closes #28

### CODE REVIEW

**Comment by rebecca-butler** [comment]
Paging @ivanpauno for review.

File: src/domain_bridge/domain_bridge.cpp
Comment (ivanpauno) [inline src/domain_bridge/domain_bridge.cpp:119]: you can delete this line, `create_generic_publisher()` already does this [here](https://github.com/ros2/rclcpp/blob/1037822a63330495dcf0de5e8f20544375a5f116/rclcpp/include/rclcpp/create_generic_publisher.hpp#L63).

IIRC, you can pass a callback group in the [options](https://github.com/ros2/rclcpp/blob/1037822a63330495dcf0de5e8f20544375a5f116/rclcpp/include/rclcpp/node.hpp#L290) argument.

File: src/domain_bridge/domain_bridge.cpp
Comment (ivanpauno) [inline src/domain_bridge/domain_bridge.cpp:139]: same here

File: src/domain_bridge/domain_bridge.cpp
Comment (rebecca-butler) [reply src/domain_bridge/domain_bridge.cpp]: Fixed.

File: src/domain_bridge/domain_bridge.cpp
Comment (rebecca-butler) [reply src/domain_bridge/domain_bridge.cpp]: I've removed this line and passed the callbacks into the options argument.

**Review by ivanpauno — APPROVED** [review]
LGTM!
I don't have write access, so waiting for @jacobperron to merge it.

P.S.: Maybe we can be given write access here (?)

**Comment by jacobperron** [comment]
@clalancette Can you give the ROS 2 team write access to this repo? thanks!

**Pull request merged**
